homeassistant/components/roon/media_player.py

```diff
@@ -0,0 +1,521 @@
+
+    def media_play_pause(self):
+        """Toggle play command to device."""
+        self._server.roonapi.playback_control(self.output_id, "playpause")
+
+    def media_stop(self):
+        """Send stop command to device."""
+        self._server.roonapi.playback_control(self.output_id, "stop")
+
+    def media_next_track(self):
+        """Send next track command to device."""
+        self._server.roonapi.playback_control(self.output_id, "next")
+
+    def media_previous_track(self):
+        """Send previous track command to device."""
+        self._server.roonapi.playback_control(self.output_id, "previous")
+
+    def media_seek(self, position):
+        """Send seek command to device."""
+        self._server.roonapi.seek(self.output_id, position)
+
+    def set_volume_level(self, volume):
+        """Send new volume_level to device."""
+        volume = int(volume * 100)
+        try:
+            self._server.roonapi.change_volume(self.output_id, volume)
+        except Exception as exc:  # pylint: disable=broad-except
+            _LOGGER.error(
+                "set_volume_level failed for entity %s \n %s", self.entity_id, str(exc)
+            )
+
+    def mute_volume(self, mute=True):
+        """Send mute/unmute to device."""
+        self._server.roonapi.mute(self.output_id, mute)
+
+    def volume_up(self):
+        """Send new volume_level to device."""
+        self._server.roonapi.change_volume(self.output_id, 3, "relative")
+
+    def volume_down(self):
+        """Send new volume_level to device."""
+        self._server.roonapi.change_volume(self.output_id, -3, "relative")
+
+    def turn_on(self):
+        """Turn on device (if supported)."""
+        if self.supports_standby and "source_controls" in self.player_data:
+            for source in self.player_data["source_controls"]:
+                if source["supports_standby"] and source["status"] != "indeterminate":
+                    self._server.roonapi.convenience_switch(
+                        self.output_id, source["control_key"]
+                    )
+                    break
+        else:
+            return self.media_play()
+
+    def turn_off(self):
+        """Turn off device (if supported)."""
+        if self.supports_standby and "source_controls" in self.player_data:
+            for source in self.player_data["source_controls"]:
+                if (
+                    source["supports_standby"]
+                    and not source["status"] == "indeterminate"
+                ):
+                    self._server.roonapi.standby(self.output_id, source["control_key"])
+                    break
+        else:
+            return self.media_stop()
+
+    def set_shuffle(self, shuffle):
+        """Set shuffle state on zone."""
+        self._server.roonapi.shuffle(self.output_id, shuffle)
+
+    def select_source(self, source):
+        """Select source on player (used to sync/unsync)."""
```

Comment:
This sounds weird. That select source service should only be used to select a source nothing else. Other tasks need to be done by custom entity services if there's no existing service that is appropriate.

Reply:
Removes for now - can always add something custom later.